From ba964124cc1e2e76cf12404f8fccbc0046ebca7c Mon Sep 17 00:00:00 2001
From: Rui Sousa <rui.sousa@nxp.com>
Date: Fri, 26 Nov 2021 14:11:39 +0100
Subject: [PATCH 13/68] configs: RTOS<->Linux intercell communication

Add definition of memory regions and PCI devices for intercell
communication. Definition is aligned with the demo device definition
in the root cell:
- 3 peers total (even if we only use two)

Signed-off-by: Rui Sousa <rui.sousa@nxp.com>
---
 configs/arm64/imx8mm-harpoon-freertos-audio.c |  5 +++++
 configs/arm64/imx8mm-harpoon-freertos.c       |  5 +++++
 configs/arm64/imx8mm-harpoon-rtos.h           | 13 +++++++++++--
 configs/arm64/imx8mn-harpoon-freertos-audio.c |  5 +++++
 configs/arm64/imx8mn-harpoon-freertos.c       |  5 +++++
 configs/arm64/imx8mn-harpoon-rtos.h           | 13 +++++++++++--
 configs/arm64/imx8mp-harpoon-freertos-audio.c |  5 +++++
 configs/arm64/imx8mp-harpoon-freertos.c       |  5 +++++
 configs/arm64/imx8mp-harpoon-rtos.h           | 13 +++++++++++--
 configs/arm64/imx8mp-harpoon-zephyr.c         |  5 +++++
 10 files changed, 68 insertions(+), 6 deletions(-)

diff --git a/configs/arm64/imx8mm-harpoon-freertos-audio.c b/configs/arm64/imx8mm-harpoon-freertos-audio.c
index a5a81d23..7149762f 100644
--- a/configs/arm64/imx8mm-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mm-harpoon-freertos-audio.c
@@ -48,4 +48,9 @@
 	/* interrupts 128..159 */      \
 	RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
diff --git a/configs/arm64/imx8mm-harpoon-freertos.c b/configs/arm64/imx8mm-harpoon-freertos.c
index 2a20580f..eda6234d 100644
--- a/configs/arm64/imx8mm-harpoon-freertos.c
+++ b/configs/arm64/imx8mm-harpoon-freertos.c
@@ -42,4 +42,9 @@
 	/* interrupts 128..159 */       \
 	RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
diff --git a/configs/arm64/imx8mm-harpoon-rtos.h b/configs/arm64/imx8mm-harpoon-rtos.h
index 925a0784..4089b923 100644
--- a/configs/arm64/imx8mm-harpoon-rtos.h
+++ b/configs/arm64/imx8mm-harpoon-rtos.h
@@ -12,13 +12,18 @@
  *
  */
 
-#define RTOS_COMMON_MEMORY_REGIONS_NUM                       5
+#define RTOS_COMMON_MEMORY_REGIONS_NUM                       10
 #define RTOS_COMMON_MEMORY_REGIONS \
 	MMIO_REGION_RW( 0x302d0000, 0x302d0000, KB(64)), /* GPT1 */                 \
 	MMIO_REGION_RW( 0x302e0000, 0x302e0000, KB(64)), /* GPT2 */                 \
 	MMIO_REGION_ROS(0x30360000, 0x30360000, KB(64)), /* ANA_PLL */              \
 	MMIO_REGION_ROS(0x30380000, 0x30380000, KB(64)), /* CCM */                  \
-	MMIO_REGION_RW( 0x30a60000, 0x30a60000, KB(64))  /* UART4 */
+	MMIO_REGION_RW( 0x30a60000, 0x30a60000, KB(64)), /* UART4 */                \
+	MEM_REGION_ROS( 0xbbaf0000, 0xbbaf0000, KB(4)),  /* IVSHMEM */              \
+	MEM_REGION_RWS( 0xbbaf1000, 0xbbaf1000, KB(36)), /* IVSHMEM */              \
+	MEM_REGION_ROS( 0xbbafa000, 0xbbafa000, KB(8)),  /* IVSHMEM */              \
+	MEM_REGION_RWS( 0xbbafc000, 0xbbafc000, KB(8)),  /* IVSHMEM */              \
+	MEM_REGION_ROS( 0xbbafe000, 0xbbafe000, KB(8))   /* IVSHMEM */
 
 #define RTOS_COMMON_IRQCHIPS_NUM                             1
 /* interrupts 32..63 */
@@ -37,3 +42,7 @@
 /* interrupts 128..159 */
 #define RTOS_COMMON_IRQCHIPS_BITMAP4 \
 	0
+
+#define RTOS_COMMON_PCI_IVSHMEM_NUM                       1
+#define	RTOS_COMMON_PCI_IVSHMEM \
+	PCI_IVSHMEM_UNDEFINED(1, 0, 5, 1, 3)
diff --git a/configs/arm64/imx8mn-harpoon-freertos-audio.c b/configs/arm64/imx8mn-harpoon-freertos-audio.c
index ea6fbd15..8e94eff9 100644
--- a/configs/arm64/imx8mn-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mn-harpoon-freertos-audio.c
@@ -48,4 +48,9 @@
 	/* interrupts 128..159 */       \
 	RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
diff --git a/configs/arm64/imx8mn-harpoon-freertos.c b/configs/arm64/imx8mn-harpoon-freertos.c
index 303bee7e..6f759f0d 100644
--- a/configs/arm64/imx8mn-harpoon-freertos.c
+++ b/configs/arm64/imx8mn-harpoon-freertos.c
@@ -42,4 +42,9 @@
 	/* interrupts 128..159 */       \
 	RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
diff --git a/configs/arm64/imx8mn-harpoon-rtos.h b/configs/arm64/imx8mn-harpoon-rtos.h
index c487e6e4..cb4e334d 100644
--- a/configs/arm64/imx8mn-harpoon-rtos.h
+++ b/configs/arm64/imx8mn-harpoon-rtos.h
@@ -12,13 +12,18 @@
  *
  */
 
-#define RTOS_COMMON_MEMORY_REGIONS_NUM                       5
+#define RTOS_COMMON_MEMORY_REGIONS_NUM                       10
 #define RTOS_COMMON_MEMORY_REGIONS \
 	MMIO_REGION_RW( 0x302d0000, 0x302d0000, KB(64)), /* GPT1 */                 \
 	MMIO_REGION_RW( 0x302e0000, 0x302e0000, KB(64)), /* GPT2 */                 \
 	MMIO_REGION_ROS(0x30360000, 0x30360000, KB(64)), /* ANA_PLL */              \
 	MMIO_REGION_ROS(0x30380000, 0x30380000, KB(64)), /* CCM */                  \
-	MMIO_REGION_RW( 0x30a60000, 0x30a60000, KB(64))  /* UART4 */
+	MMIO_REGION_RW( 0x30a60000, 0x30a60000, KB(64)), /* UART4 */                \
+	MEM_REGION_ROS( 0xbbaf0000, 0xbbaf0000, KB(4)),  /* IVSHMEM */              \
+	MEM_REGION_RWS( 0xbbaf1000, 0xbbaf1000, KB(36)), /* IVSHMEM */              \
+	MEM_REGION_ROS( 0xbbafa000, 0xbbafa000, KB(8)),  /* IVSHMEM */              \
+	MEM_REGION_RWS( 0xbbafc000, 0xbbafc000, KB(8)),  /* IVSHMEM */              \
+	MEM_REGION_ROS( 0xbbafe000, 0xbbafe000, KB(8))   /* IVSHMEM */
 
 #define RTOS_COMMON_IRQCHIPS_NUM                             1
 /* interrupts 32..63 */
@@ -37,3 +42,7 @@
 /* interrupts 128..159 */
 #define RTOS_COMMON_IRQCHIPS_BITMAP4 \
 	0
+
+#define RTOS_COMMON_PCI_IVSHMEM_NUM                       1
+#define	RTOS_COMMON_PCI_IVSHMEM \
+	PCI_IVSHMEM_UNDEFINED(0, 0, 5, 1, 3)
diff --git a/configs/arm64/imx8mp-harpoon-freertos-audio.c b/configs/arm64/imx8mp-harpoon-freertos-audio.c
index 6df3c313..6190a2ea 100644
--- a/configs/arm64/imx8mp-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mp-harpoon-freertos-audio.c
@@ -54,4 +54,9 @@
 	/* interrupts 128..159 */       \
 	RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
diff --git a/configs/arm64/imx8mp-harpoon-freertos.c b/configs/arm64/imx8mp-harpoon-freertos.c
index 9ab82907..b9155ec7 100644
--- a/configs/arm64/imx8mp-harpoon-freertos.c
+++ b/configs/arm64/imx8mp-harpoon-freertos.c
@@ -42,4 +42,9 @@
 	/* interrupts 128..159 */       \
 	RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
diff --git a/configs/arm64/imx8mp-harpoon-rtos.h b/configs/arm64/imx8mp-harpoon-rtos.h
index a7f13350..dc8ae126 100644
--- a/configs/arm64/imx8mp-harpoon-rtos.h
+++ b/configs/arm64/imx8mp-harpoon-rtos.h
@@ -11,13 +11,18 @@
  *
  */
 
-#define RTOS_COMMON_MEMORY_REGIONS_NUM                       5
+#define RTOS_COMMON_MEMORY_REGIONS_NUM                       10
 #define RTOS_COMMON_MEMORY_REGIONS \
 	MMIO_REGION_RW( 0x302d0000, 0x302d0000, KB(64)), /* GPT1 */                 \
 	MMIO_REGION_RW( 0x302e0000, 0x302e0000, KB(64)), /* GPT2 */                 \
 	MMIO_REGION_ROS(0x30360000, 0x30360000, KB(64)), /* ANA_PLL */              \
 	MMIO_REGION_ROS(0x30380000, 0x30380000, KB(64)), /* CCM */                  \
-	MMIO_REGION_RW( 0x30a60000, 0x30a60000, KB(64))  /* UART4 */
+	MMIO_REGION_RW( 0x30a60000, 0x30a60000, KB(64)), /* UART4 */                \
+	MEM_REGION_ROS( 0xfd9f0000, 0xfd9f0000, KB(4)),  /* IVSHMEM */              \
+	MEM_REGION_RWS( 0xfd9f1000, 0xfd9f1000, KB(36)), /* IVSHMEM */              \
+	MEM_REGION_ROS( 0xfd9fa000, 0xfd9fa000, KB(8)),  /* IVSHMEM */              \
+	MEM_REGION_RWS( 0xfd9fc000, 0xfd9fc000, KB(8)),  /* IVSHMEM */              \
+	MEM_REGION_ROS( 0xfd9fe000, 0xfd9fe000, KB(8))   /* IVSHMEM */
 
 #define RTOS_COMMON_IRQCHIPS_NUM                             1
 /* interrupts 32..63 */
@@ -36,3 +41,7 @@
 /* interrupts 128..159 */
 #define RTOS_COMMON_IRQCHIPS_BITMAP4 \
 	0
+
+#define RTOS_COMMON_PCI_IVSHMEM_NUM                       1
+#define	RTOS_COMMON_PCI_IVSHMEM \
+	PCI_IVSHMEM_UNDEFINED(2, 0, 5, 1, 3)
diff --git a/configs/arm64/imx8mp-harpoon-zephyr.c b/configs/arm64/imx8mp-harpoon-zephyr.c
index 658d9426..c1d48024 100644
--- a/configs/arm64/imx8mp-harpoon-zephyr.c
+++ b/configs/arm64/imx8mp-harpoon-zephyr.c
@@ -41,5 +41,10 @@
 				/* interrupts 128..159 */      \
 				RTOS_COMMON_IRQCHIPS_BITMAP4
 
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_NUM	RTOS_COMMON_PCI_IVSHMEM_NUM
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
 #include "cell-create.h"
 
-- 
2.34.1

